components/tidb_query/src/codec/row_codec.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::codec::{table, Datum, Result};
+
+pub mod v_128;
+
+pub fn encode(datums: Vec<Datum>, col_ids: &[i64], version: u8) -> Result<Vec<u8>> {
```

Comment:
Why do we need encoders? I think Coprocessor only do decoding.

Reply:
Maybe we need the encoder if we push the `Update` down to coprocessor in the future.